src/smc-webapp/project_new.cjsx

```diff
@@ -238,8 +243,10 @@ ProjectNewForm = rclass ({name}) ->
         e.preventDefault()
         if @state.filename[@state.filename.length - 1] == '/'
             @create_folder()
-        else
+        else if @state.filename.includes(".")
```

Comment:
no -- There are tons of extensions that aren't explicitly listed anywhere (or known to me), but are not "invalid extensions".  

Maybe a warning would be good though, just saying we have no special support for editing that file type.  Maybe.